projects/fourier/lib/fourier/utilities/swift_package_manager.rb

```diff
@@ -83,6 +105,34 @@ def self.build_fat_release_binary(
           File.join(swift_build_directory, "#{X86_64_TARGET}/release/#{binary_name}")
         )
       end
+
+      private
+        def self.path_to_xcode(version = nil)
```

Comment:
Wouldn't be better to do this conversion early when parsing the arguments, and after that always assume we are dealing with Xcode paths?

Reply:
Yeah, I could try that. I can't decide if I agree that it would be better or not, but it's worth a try to see.

Reply:
I think yours was the right call here. I moved the parsing of the passed in version [earlier in the chain](https://github.com/tuist/tuist/pull/3957/files#diff-9a56a4817ae09cfdd18e1712b6dabb1719d00fccabf4da775aa59c7e0721725fR41-R42) (it happens right as the arguments come into the program from the command line), and also moved the logic for finding and switching Xcode versions into [its own file](https://github.com/tuist/tuist/pull/3957/files#diff-142bcd26d51d4a9edf088d96477700bc119aa18561edd172dd3aba50b3066a26).